### PR TITLE
network: handle autoconnections policy for rhel upstream

### PR DESCRIPTION
**Pull request by rvykydal**
The added shared service will be run only on rhel systems.

Resolves: RHEL-53196

### CODE REVIEW

File: data/systemd/anaconda-nm-disable-autocons-rhel.service [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+ConditionKernelCommandLine=|ip
+ConditionKernelCommandLine=|inst.ks
+ConditionOSRelease=ID=rhel
+Description=NetworkManager autoconnections configuration for Anaconda installation environment for RHEL
+Before=NetworkManager.service
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/anaconda-nm-disable-autocons
+
```
Comment (KKoukiou) [inline data/systemd/anaconda-nm-disable-autocons-rhel.service:11]: Extra new line not needed. Feel free to ignore.

**Review by KKoukiou — APPROVED** [review]

File: data/systemd/anaconda-nm-disable-autocons-rhel.service (hunk 1 above)
Comment (rvykydal) [reply data/systemd/anaconda-nm-disable-autocons-rhel.service]: Fixed.

**Review by KKoukiou — APPROVED** [review]

**Comment by jkonecny12** [comment]
I just wonder (not part of this PR). Shouldn't be the `anaconda-nm-disable-autocons` script in `/usr/libexec` instead of `/usr/bin`?

File: data/systemd/anaconda-nm-disable-autocons-rhel.service [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+ConditionKernelCommandLine=|ip
+ConditionKernelCommandLine=|inst.ks
+ConditionOSRelease=ID=rhel
```
Comment (jstodola) [inline data/systemd/anaconda-nm-disable-autocons-rhel.service:4]: This works for RHEL, unfortunately not for CentOS Stream, see https://issues.redhat.com/browse/RHEL-67815

**Pull request merged**